soc: mspm0: add power management support

SanjayyyV opened the pull request:
- Added power states for MSPM0
- Added timing details for G and L series.

SanjayyyV committed:
dts: arm/ti/mspm0: add cpu power states for MSPM0 series

add range of cpu power states (run/sleep, stop and standby) for MSPM0
series. add exit latencies ranging from 1.5-15.7µs and minimum
residency times of 5-10ms for optimal power management transitions.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce commented on soc/ti/mspm0/common/poweroff.c:
separate commit for poweroff

SanjayyyV committed:
soc: mspm0: add power management support

TI MSPM0 series supports range of power modes (RUN/SLEEP, STOP, STANDBY)
supporting low power operations. Provides automatic restoration to
RUN mode on wakeup from any low power state.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

SanjayyyV committed:
soc: mspm0: add poweroff support

add support for SHUTDOWN operating mode in TI MSPM0 series for power-off
operation. Uses HWINFO for reset cause detection to handle shutdown IO
release on low power wakeup.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce commented on soc/ti/mspm0/CMakeLists.txt:
```diff
@@ -8,3 +8,6 @@ if(CONFIG_SOC_FAMILY_TI_MSPM0)
 endif()
+
+zephyr_sources_ifdef(CONFIG_PM common/power.c)
+zephyr_sources_ifdef(CONFIG_POWEROFF common/poweroff.c)
```
this needs to be moved to poweroff commit

parthitce commented on soc/ti/mspm0/mspm0g/Kconfig:
```diff
@@ -13,5 +13,7 @@ config SOC_SERIES_MSPM0G
 	select BUILD_OUTPUT_HEX
 	select HAS_MSPM0_SDK
+	select HAS_PM
+	select HAS_POWEROFF
```
this as well to poweroff commit

parthitce commented on soc/ti/mspm0/mspm0l/Kconfig:
```diff
@@ -13,5 +13,7 @@ config SOC_SERIES_MSPM0L
 	select BUILD_OUTPUT_HEX
 	select HAS_MSPM0_SDK
+	select HAS_PM
+	select HAS_POWEROFF
```
ditto

parthitce commented:
@SanjayyyV https://github.com/linumiz/zephyr/pull/155 consider this PR as well for upstream?

parthitce reviewed (approved):
Run compliance and checkpatch stuff if not done. Otherwise LGTM.

SanjayyyV committed:
drivers: gpio: Add GPIO Fast Wake support

The fast wake feature in the MSPM0 GPIO peripheral allows the GPIO module
to stay in a low-power state and detect interrupt events on the device pins
without requiring a high-speed clock. This allows the device to support
fast wakeup from low-power modes, such as STOP and STANDBY, on any GPIO
pin.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

SanjayyyV committed:
west.yml: Update hal_ti revision

Update hal_ti to include input-enable property for all UART RX pins.
Without the input-enable property, UART reception does not function
correctly when the SoC enters low-power mode. Adding this property
ensures reliable UART operation in all power states.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>

parthitce commented:
Addressed in upstream with https://github.com/zephyrproject-rtos/zephyr/pull/94834. Closing this PR.
